src/plugins/layout/shadowdom.ts

```diff
@@ -69,59 +63,55 @@ export class ShadowDom {
 
     assert(!!parent, "moveShadowNode", "parent is missing");
```

Comment:
I think it's ok to keep them, because they don't necessarily cause critical issues, but rather lets us know that there is something wrong with the logic. So, if assertion fails, but DOM still arrives to consistent state, then we can continue collecting data safely. Otherwise, we'll arrive to the inconsistent state and stop processing mutations

---
In reply to: [132226650](https://github.com/Microsoft/clarity-js/pull/25#discussion_r132226650) [](ancestors = 132226650)